src/shared/Core/Trace2.cs

```diff
@@ -255,6 +271,34 @@ public void Stop(int exitCode, string filePath, int lineNumber)
         });
     }
 
+    public void WriteError(
+        string errorMessage,
+        string formattedErrorMessage = null,
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0)
+    {
+        // It is possible for an error to be thrown before TRACE2 can be initialized.
+        // Since certain dependencies are not available until initialization,
+        // we must immediately return if this method is invoked prior to
+        // initialization.
+        if (!_initialized)
+        {
+            return;
+        }
+
+        WriteMessage(new ErrorMessage()
+        {
+            Event = Trace2Event.Error,
+            Sid = _sid,
+            Time = DateTimeOffset.UtcNow,
+            File = Path.GetFileName(filePath).ToLower(),
+            Line = lineNumber,
+            Message = errorMessage,
+            FormattedErrorMessage = formattedErrorMessage ?? "",
```

Comment:
Given the C# model of building message strings is different from C, would it be better to default the format string to the message string if the format string is not supplied?

With the assumption being that your call sites in the next commit will send one string if there's no PII/sensitive information (and you can build the message the usual way with inline @"{var}" tokens).  (More on this in the next commit.)

And only if there is something that we think is sensitive, would we use the 2 arg version of `Trace2.WriteError()`.

The original purpose of the "fmt" and "msg" field pair was that we could aggregate errors of the form "invalid pathname: %s" in SQL/kusto/wherever on the "fmt" field without using regex magic -- in addition to the privacy aspects, since the actual pathname might be sensitive.

Reply:
Hmm...if we do this, won't the error message often be printed twice? Since it will be the same as the format string?